packages/amplify_auth_cognito/example/integration_test/get_current_user_test.dart

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+
+import 'utils/mock_data.dart';
+import 'utils/setup_utils.dart';
+import 'utils/validation_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  final username = generateUsername();
+  final password = generatePassword();
+
+  group('getCurrentUser', () {
+    setUpAll(() async {
+      await configureAuth();
+
+      // create one user for all tests
+      await Amplify.Auth.signUp(
+          username: username,
+          password: password,
+          options: CognitoSignUpOptions(userAttributes: {
+            'email': generateEmail(),
+            'phone_number': mockPhoneNumber
+          }));
+    });
+
+    // sign in prior to each test
+    setUp(() async {
+      await signOutUser();
+      await Amplify.Auth.signIn(
+        username: username,
+        password: password,
+      );
+    });
+
+    testWidgets('should return the current user', (WidgetTester tester) async {
+      var authUser = await Amplify.Auth.getCurrentUser();
+      // usernames need to be compared case insensitive due to
+      // https://github.com/aws-amplify/amplify-flutter/issues/723
+      expect(authUser.username.toLowerCase(), username.toLowerCase());
+      expect(isValidUserSub(authUser.userId), isTrue);
+    });
+
+    testWidgets('should throw SignedOutException if the user is signed out',
+        (WidgetTester tester) async {
+      await Amplify.Auth.signOut();
+      try {
+        await Amplify.Auth.getCurrentUser();
+      } catch (e) {
+        expect(e, TypeMatcher<SignedOutException>());
+        return;
+      }
+      throw Exception('Expected SignedOutException');
```

Comment:
kinda dumb question: Is it better to throw exception like this or call `fail()`? Can we confirm the test will fail and the task returns non 0 exit status when this exception is thrown? It would seem obvious that it would, just I've just noticed weird stuff with the integration test driver related to exceptions/failures, so just wanted to confirm.

I realize this pattern already exists in other integ tests that I approved. If this makes the test fail, as expected, then this pattern is fine IMO and `fail()` has no advantage.

Reply:
It will fail the tests, but `fail()` is a little clearer to understand the intention of, which is probably the reason it exists. Updated 👍